packages/synth.queues/src/services/kafka.queue.service.ts

```diff
@@ -0,0 +1,118 @@
+import { PluginManager } from '@applicature/synth.plugin-manager';
+import * as kafka from 'node-rdkafka';
+import * as logger from 'winston';
+import { KafkaConfig, Message, Queue } from '../types';
+import { QueueService } from './queue.service';
+export class KafkaQueueService extends QueueService {
+    private producerReady: any;
+    private consumerReady: any;
+    private adminClient: kafka.InternalAdminClient;
+    private topics: Set<string>;
+
+    // tslint:disable-next-line:no-shadowed-variable
+    constructor(pluginManager: PluginManager, config: KafkaConfig) {
+        super(pluginManager);
+
+        const producer = new kafka.Producer(config.producerConfig, config.topicConfig);
```

Comment:
move initialization of Producer, KafkaConsumer, AdminClient to init method, because they use async operation for initialization